### ISSUE TITLE
Calling `wrapper` task overwrites the gradle distribution URL when using a mirror

### ISSUE THREAD

**Issue by tudortimi** [gradle/gradle#28209]
### Issue type

Polishing

### Problem description

In my team, we use custom repos to download Gradle distributions. Let's say that we already have a Gradle wrapper, with version 6.7.1. If we want to upgrade to 7.4.2 we would do:

```
./gradle wrapper --gradle-distribution-url https\:/.../artifactory/gradle-distributions/gradle-7.4.2-bin.zip
```

Now, `gradle-wrapper.properties` has the new URL. Running `./gradlew` would start version 7.4.2.

The wrapper script itself would still contain the code from 6.7.1. To update the script, we'd have to run the `wrapper` task again. If we just call `./gradlew`, Gradle will overwrite the URL in `gradle-wrapper.properties` with `https\://services.gradle.org/distributions/gradle-7.4.2-bin.zip` (the same version, but not downloaded from the custom repo anymore). This is annoying, as the user has to remember to pass the same URL to the `wrapper` task for the second call.

**Comment by ov7a** [comment]
Your issue appears to be a question about how to use Gradle. They should be asked in a different channel, so we will close it. Your question might be answered in Gradle documentation or on our community Slack or forums. Find links to these resources on [help.gradle.org](https://help.gradle.org/).

If you think our analysis is wrong, please provide us with more detailed information explaining why.

***

You can solve this by configuring the distribution URL in the gradle build. See the [documentation](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.wrapper.Wrapper.html).

Related:
* https://github.com/gradle/gradle/issues/27295

**Comment by ov7a** [comment]
> Do I understand correctly from the docs that I can set a "template" String for the distributionUrl property of the wrapper task? This would help at least. I'm guessing it isn't, though, because otherwise https://github.com/gradle/gradle/issues/27295 wouldn't exist.

Unfortunately, it's not, the docs are misleading. This is something that can be solved in #27295 (probably it is the preferable way of resolving this)
